src/bls/bls_aggregator.cpp

```diff
@@ -292,47 +337,80 @@ uint64_t bls_aggregator::nodes_request(
     const size_t MAX_CONNECTIONS = 900;
 
     // FIXME: make this function async rather than blocking
-
     std::vector<service_nodes::service_node_address> snodes;
-    core.service_node_list.copy_reachable_active_service_node_addresses(
+    core.service_node_list.copy_reachable_service_node_addresses(
             std::back_inserter(snodes), core.get_nettype());
 
     auto& omq = core.omq();
     for (size_t i = 0; i < snodes.size(); i++) {
         auto& snode = snodes[i];
-        if (1) {
-            std::lock_guard connection_lock(connection_mutex);
-            ++active_connections;
+        if (core.service_node_list.is_service_node(snode.sn_pubkey, /*require_active*/ false)) {
+            if (1) {
+                std::lock_guard connection_lock(connection_mutex);
+                ++active_connections;
+            } else {
+                // TODO(doyle): Rate limit
+                std::unique_lock connection_lock(connection_mutex);
+                cv.wait(connection_lock,
+                        [&active_connections] { return active_connections < MAX_CONNECTIONS; });
+            }
+
+            omq.request(
+                    tools::view_guts(snode.x_pubkey),
+                    request_name,
+                    [i, &snodes, &connection_mutex, &active_connections, &cv, &callback](
+                            bool success, std::vector<std::string> data) {
+                        callback(bls_response{snodes[i], success}, data);
+                        std::lock_guard connection_lock{connection_mutex};
+                        assert(active_connections);
+                        if (--active_connections == 0)
+                            cv.notify_all();
+                    },
+                    message);
         } else {
-            // TODO(doyle): Rate limit
-            std::unique_lock connection_lock(connection_mutex);
-            cv.wait(connection_lock,
-                    [&active_connections] { return active_connections < MAX_CONNECTIONS; });
+            // TODO: I'm unable to get this to work, because, I believe SN's reject
+            // non-authenticated requests.
+            #if 0
+            if (1) {
+                std::lock_guard connection_lock(connection_mutex);
+                ++active_connections;
+            } else {
+                // TODO(doyle): Rate limit
+                std::unique_lock connection_lock(connection_mutex);
+                cv.wait(connection_lock,
+                        [&active_connections] { return active_connections < MAX_CONNECTIONS; });
+            }
+
+            auto addr = oxenmq::address{"tcp://{}:{}"_format(
+                    epee::string_tools::get_ip_string_from_int32(snode.ip), snode.port, tools::view_guts(snode.x_pubkey))};
```

Comment:
Right now this is just throwing away the pubkey argument as an unused format argument, so the address ends up indicating a plaintext connection rather than an encrypted one.  Either of these alternatives should solve it:
```suggestion
            auto addr = oxenmq::address{"curve://{}:{}/{:x}"_format(
                    epee::string_tools::get_ip_string_from_int32(snode.ip), snode.port, snode.x_pubkey)};
```
or:
```suggestion
            auto addr = oxenmq::address{"tcp://{}:{}"_format(
                    epee::string_tools::get_ip_string_from_int32(snode.ip), snode.port)), tools::view_guts(snode.x_pubkey)};
```